### PR TITLE
Validations

### PR DESCRIPTION
**Pull request by skrix**

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -10,5 +10,13 @@
   context 'with invalid attributes' do
     it { should validate_presence_of(:password) }
     it { should validate_presence_of(:email) }
+    it { should validate_presence_of(:username) }
+
+    it { should validate_uniqueness_of(:email).case_insensitive }
+    it { should validate_uniqueness_of(:username) }
+
+    it do
+      should validate_length_of(:username).is_at_least(2).is_at_most(30)
```
[the remark this answers is not in the document]
Comment (ailecksandr) [reply spec/models/user_spec.rb]: this way you don't need to change test after changing constant

File: spec/models/item_spec.rb [hunk 2]
```diff
@@ -16,5 +16,12 @@
     it { should validate_presence_of(:price) }
     it { should validate_presence_of(:kind) }
     it { should have_many(:meals) }
+
+    it { should validate_numericality_of (:price) }
+    it { should validate_uniqueness_of(:name) }
+
+    it do
+      should validate_length_of(:name).is_at_least(2).is_at_most(30)
```
Comment (ailecksandr) [inline spec/models/item_spec.rb]: same

File: app/models/item.rb [hunk 3]
```diff
@@ -5,6 +5,9 @@ class Item < ApplicationRecord
   has_many :menus, through: :meals
 
   validates :name, :price, :kind, presence: true
+  validates :name, uniqueness: true
+  validates :name, length: { in: User::NAME_LENGTH }
```
Comment (volodialeobit) [inline app/models/item.rb]: align

File: spec/models/item_spec.rb [hunk 4]
```diff
@@ -16,5 +16,12 @@
     it { should validate_presence_of(:price) }
     it { should validate_presence_of(:kind) }
     it { should have_many(:meals) }
+
+    it { should validate_numericality_of (:price) }
```
Comment (volodialeobit) [inline spec/models/item_spec.rb]: remove space

File: spec/models/meal_spec.rb [hunk 5]
```diff
@@ -27,5 +27,6 @@
     it { should validate_presence_of(:price) }
     it { should belong_to(:item) }
     it { should belong_to(:menu) }
+    it { should validate_numericality_of (:price) }
```
Comment (volodialeobit) [inline spec/models/meal_spec.rb]: remove space

File: spec/models/menu_spec.rb [hunk 6]
```diff
@@ -9,5 +9,6 @@
 
   context 'with invalid attributes' do
     it { should have_many(:meals) }
+    it { should accept_nested_attributes_for (:meals) }
```
Comment (volodialeobit) [inline spec/models/menu_spec.rb]: remove space

File: spec/models/order_spec.rb [hunk 7]
```diff
@@ -8,4 +8,9 @@
   context 'with valid params' do
     it { expect(order).to be_valid }
   end
+
+  context 'with invalid attributes' do
+    it { should belong_to(:user) }
+    it { should accept_nested_attributes_for (:order_meals) }
```
Comment (volodialeobit) [inline spec/models/order_spec.rb]: remove space

**Pull request merged**